ydb/library/yql/core/common_opt/yql_co_simple1.cpp

```diff
@@ -4665,6 +4665,16 @@ void RegisterCoSimpleCallables1(TCallableOptimizerMap& map) {
     map["IsDistinctFrom"] = std::bind(&OptimizeDistinctFrom<false>, _1, _2);
 
     map["StartsWith"] = map["EndsWith"] = map["StringContains"] = [](const TExprNode::TPtr& node, TExprContext& ctx, TOptimizeContext& /*optCtx*/) {
+        if (node->Head().GetTypeAnn()->GetKind() == ETypeAnnotationKind::Pg || node->Tail().GetTypeAnn()->GetKind() == ETypeAnnotationKind::Pg) {
```

Comment:
why this overload is needed?

Reply:
This overload helps to simplify logic of predicate analysis. This way we make StartsWith with pg types to look exactly the same as with plain YQL types (no intermediate FromPg is needed), so the code which looks for StartsWith over Member() call will work without pg-specific tweaks.

Also, accepting PG types here looks like a nice feature to me.